docs/Developers/Executing_Tests.rst

```diff
@@ -56,8 +56,14 @@ To run our tests, you can also use ``python3 setup.py test``
 
 .. note::
 
-    You will not get a test coverage of 100% - the coverage on the
-    website is merged for several python versions.
+    You may not get a test coverage of 100% locally. The coverage
+    published on `codecov.io <https://codecov.io/gh/coala/>`__ (GitHub
+    Projects) and `codecov.io <https://codecov.io/gl/coala/>`__ (GitLab
+    Projects) are actually merged results for several python versions.
+    Also, the results are merged from different OS. Appveyor results
+    provide coverage of Windows specific lines, and Travis/Circle
+    provide coverage of Unix specific lines. Thus, your test coverage
+    can 'pass' without reaching 100% .
```

Comment:
this doesn't seem like the correct explanation for why the coverage is not full.
its due to 
* pragma no cover
* lack of tests, I stumble upon writing bad code and still passing tests every now and then and reviews correcting me for it. I assume everything in coala is not tested and the initial part of it was merged without adding tests. should confirm this by a maintainer.

Reply:
whatever explanation gets the job done 👍
but it shouldn't be wrong

Reply:
@jayvdb 
>Also, the results are merged from different OS

don't know if that's true, each one produces separate coverage report for its own environment. even different versions of python

Reply:
The info that @ishanSrt is providing is also very important, as developers often forget to write tests, and that is typically why they will see test coverage not reach 100%.  But for that to happen, they wont have run `pytest`, because it will load the plugins mentioned in `setup.cfg`, which includes coverage, and it will fail locally if they havent written the tests yet.

The solution is not for the developer to add `#pragma: no cover`.  The code reviewer will usually reject that.  And soon we'll have CI automatically reject that, except in the most limited of cases.

Usually the correct solution is write some tests.

Reply:
how is getting a merged coverage report (that is if we get one) enabling the test coverage to pass even when its not 100%